server/lib/parser/transaction.js

```diff
@@ -16,59 +16,20 @@ function parse(entry, txs) {
     counter++;
 
     if (counter % socketThrottle === 0) {
+
+      counter = 0;
+
       io.sockets.emit('tx', {
         txid: txJSON.hash,
-        valueOut: tx.outputs.reduce((a, b) => {
-          a = a.toJSON();
-          b = b.toJSON();
+        valueOut: tx.outputs.reduce((sum, tx) => {
+          tx = tx.toJSON();
 
-          const valA = (a.value || a.valueOut.value || 0) / 1e8;
-          const valB = (b.value || b.valueOut.value || 0) / 1e8;
+          const valB = (tx.value || tx.valueOut.value || 0) / 1e8;
 
-          return valA + valB;
-        }),
-      }, 0);
+          return sum + valB;
+        }, 0),
+      });
     }
-
-    const t = new TxModel({
```

Comment:
I don't think you'll be able to index that - not positive, but pretty sure that multikey indexes can only descend through one layer of array nesting - https://docs.mongodb.com/manual/core/index-multikey/#index-arrays-with-embedded-documents . So i believe it will break down if you try to index block.txs.inputs.address , since both txs and inputs are arrays. If it will that's great, but you should verify that such an index can be added and used before continuing down that path.